examples/api-call-example/components/with-auth.jsx

```diff
@@ -1,42 +1,41 @@
-import React, { Component } from 'react';
+import React from 'react';
+import { useUser } from '@auth0/nextjs-auth0';
 
 import auth0 from '../lib/auth0';
-import { fetchUser } from '../lib/user';
 import createLoginUrl from '../lib/url-helper';
 import RedirectToLogin from '../components/login-redirect';
 
 export default function withAuth(InnerComponent) {
-  return class Authenticated extends Component {
-    static async getInitialProps(ctx) {
-      if (!ctx.req) {
-        const user = await fetchUser();
-        return {
-          user
-        };
-      }
-
-      const session = await auth0.getSession(ctx.req, ctx.res);
-      if (!session || !session.user) {
-        ctx.res.writeHead(302, {
-          Location: createLoginUrl(ctx.req.url)
-        });
-        ctx.res.end();
-        return;
-      }
-
-      return { user: session.user };
+  const Authenticated = (props) => {
+    const { user } = useUser();
+
+    if (!user) {
+      return <RedirectToLogin />; // do you need a "redirecting to login" route?
     }
 
-    constructor(props) {
-      super(props);
+    return <InnerComponent {...props} user={user} />;
+  };
+
+  Authenticated.getInitialProps = async (ctx) => {
+    if (!ctx.req) {
+      const response = await fetch('/api/me');
+      const result = response.ok ? await response.json() : null;
+
+      return { user: result };
     }
 
-    render() {
-      if (!this.props.user) {
-        return <RedirectToLogin />;
-      }
+    const session = await auth0.getSession(ctx.req, ctx.res);
```

Comment:
The TS type for `getSession` only takes the request, but if the response is not passed it fails at runtime.